x-pack/plugins/reporting/server/export_types/common/pdf/pdfmaker.ts

```diff
@@ -5,52 +5,70 @@
  * 2.0.
  */
 
-import { i18n } from '@kbn/i18n';
-// @ts-ignore: no module definition
-import concat from 'concat-stream';
+import { SerializableRecord } from '@kbn/utility-types';
 import _ from 'lodash';
 import path from 'path';
-import Printer from 'pdfmake';
+import { Worker } from 'worker_threads';
 import { Content, ContentImage, ContentText } from 'pdfmake/interfaces';
 import type { Layout } from '../../../../../screenshotting/server';
-import { getDocOptions, REPORTING_TABLE_LAYOUT } from './get_doc_options';
+import { REPORTING_TABLE_LAYOUT } from './get_doc_options';
 import { getFont } from './get_font';
-import { getTemplate } from './get_template';
-
-const assetPath = path.resolve(__dirname, '..', '..', 'common', 'assets');
-const tableBorderWidth = 1;
+import {
+  headingHeight,
+  pageMarginBottom,
+  pageMarginTop,
+  pageMarginWidth,
+  subheadingHeight,
+  tableBorderWidth,
+} from './constants';
+import { PdfWorkerOutOfMemoryError } from './pdf_generate_errors';
+
+import type { PdfWorkerData } from './worker';
 
 export class PdfMaker {
-  private _layout: Layout;
+  _layout: Layout;
   private _logo: string | undefined;
   private _title: string;
   private _content: Content[];
-  private _printer: Printer;
-  private _pdfDoc: PDFKit.PDFDocument | undefined;
 
-  constructor(layout: Layout, logo: string | undefined) {
-    const fontPath = (filename: string) => path.resolve(assetPath, 'fonts', filename);
-    const fonts = {
-      Roboto: {
-        normal: fontPath('roboto/Roboto-Regular.ttf'),
-        bold: fontPath('roboto/Roboto-Medium.ttf'),
-        italics: fontPath('roboto/Roboto-Italic.ttf'),
-        bolditalics: fontPath('roboto/Roboto-Italic.ttf'),
-      },
-      'noto-cjk': {
-        // Roboto does not support CJK characters, so we'll fall back on this font if we detect them.
-        normal: fontPath('noto/NotoSansCJKtc-Regular.ttf'),
-        bold: fontPath('noto/NotoSansCJKtc-Medium.ttf'),
-        italics: fontPath('noto/NotoSansCJKtc-Regular.ttf'),
-        bolditalics: fontPath('noto/NotoSansCJKtc-Medium.ttf'),
-      },
-    };
+  private worker?: Worker;
+
+  protected workerModulePath = path.resolve(__dirname, './worker.js');
+
+  /**
+   * The maximum heap size for old memory region of the worker thread.
+   *
+   * @note We need to choose a sane number given that we need to load a Kibana
+   * node environment, all the library code and the subsequent buffers that
```

Comment:
The worker thread isn't actually running a "Kibana node environment," right? It is not using Kibana dependencies like the ES client or security. And, it wouldn't be able to, since the interface between the thread is for serializable data. Is this correct?